### ISSUE TITLE
[BUG] httputil.ReverseProxy{ FlushInterval }  CANNOT pass in negative value as docs tell

### ISSUE THREAD

**Issue by lnshi** [golang/go#31069]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.11.5 darwin/amd64
</pre>

### Does this issue reproduce with the latest release?
Haven't tried yet.

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/i***/Library/Caches/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/i***/go_projects"
GOPROXY=""
GORACE=""
GOROOT="/usr/local/Cellar/go/1.11.5/libexec"
GOTMPDIR=""
GOTOOLDIR="/usr/local/Cellar/go/1.11.5/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/xw/fjl2c85x0psg7nm1sth8y_m40000gn/T/go-build105484579=/tmp/go-build -gno-record-gcc-switches -fno-common"

</pre></details>

### What did you do?

I tried to pass in negative value(`-1` or `-1ms time.Duration`) for the [ReverseProxy.FlushInterval](https://golang.org/pkg/net/http/httputil/#ReverseProxy) as the docs tell:

> // FlushInterval specifies the flush interval
        // to flush to the client while copying the
        // response body.
        // If zero, no periodic flushing is done.
        **// A negative value means to flush immediately
        // after each write to the client.**
        // The FlushInterval is ignored when ReverseProxy
        // recognizes a response as a streaming response;
        // for such responses, writes are flushed to the client
        // immediately.
        FlushInterval time.Duration

Then server can start, but during the runtime(actual API call) the server crashes with error:
```
panic: non-positive interval for NewTicker

goroutine 66 [running]:
time.NewTicker(0xfffffffffff0bdc0, 0x1e6f410)
	/usr/local/Cellar/go/1.11.5/libexec/src/time/tick.go:23 +0x18f
net/http/httputil.(*maxLatencyWriter).flushLoop(0xc0003325d0)
	/usr/local/Cellar/go/1.11.5/libexec/src/net/http/httputil/reverseproxy.go:419 +0x44
created by net/http/httputil.(*ReverseProxy).copyResponse
	/usr/local/Cellar/go/1.11.5/libexec/src/net/http/httputil/reverseproxy.go:343 +0x2a6
exit status 2
```

### What did you expect to see?

As the docs tell:
> // A negative value means to flush immediately
        // after each write to the client.

### What did you see instead?
The server crashes upon the API call.

**Comment by lnshi** [comment]
@googlebot this has nothing to do with documentation, this is a bug, call someone to help to investigate!

**Comment by agnivade** [comment]
This is fixed in 1.12. Please feel free to reopen if it still happens for you in 1.12.
